### PR TITLE
Remove redundant messaging from uaclient

### PR DESCRIPTION
**Pull request by lucasmoura**
## Proposed Commit Message
Currently, uaclient is advertising esm services by allowing motd to show messages saying that additional packages could be installed if esm service was enabled. We don't need those type of messages because update-notifier will handle them.

Also, we are updating the code to always show expired message for esm-infra services, even if the distro is not ESM

Message in update-notifer:
https://git.launchpad.net/update-notifier/tree/data/apt_check.py#n119

## Test Steps
Run the modified integration test

## Desired commit type::
Remove redundant messaging from uaclient

## Checklist:

 - [ ] I have updated or added any unit tests accordingly
 - [ ] I have updated or added any integration tests accordingly
 - [ ] I have updated or added any documentation accordingly

### CODE REVIEW

File: apt-hook/hook.cc [hunk 1]
```diff
@@ -249,6 +249,9 @@ static void process_template_file(
          message_static_file.close();
       }
    }
+   else {
```
Comment (blackboxsw) [inline apt-hook/hook.cc:252]:
```suggestion
   } else {
```

**Review by orndorffgrant — APPROVED** [review]
This aligns with what we discussed earlier, all tests passed, and my manual test of removing static files if template isn't here worked. LGTM!

File: lib/ua_update_messaging.py [hunk 2]
```diff
@@ -225,7 +216,14 @@ def write_apt_and_motd_templates(cfg: config.UAConfig, series: str) -> None:
...
         )
-    if util.is_active_esm(series):
+
+    # We only have esm-infra apt alerts for esm distros.
+    # However, if we have expired credentials, we will
+    # produce esm-infra message showing that the contract is
+    # expiring/expired.
+    infra_status, _ = infra_inst.application_status()
+    is_infra_enabled = infra_status == ApplicationStatus.ENABLED
+    if is_infra_enabled or util.is_active_esm(series):
```
Comment (blackboxsw) [inline lib/ua_update_messaging.py:226]: Looks good.

**Review by blackboxsw — APPROVED** [review]
apt-hook looks good, we need an additional set of checks though I think to avoid writing the aggregated output message files
Here's a diff
```diff

diff --git a/apt-hook/hook.cc b/apt-hook/hook.cc
index 2298326d..73e1ba77 100644
--- a/apt-hook/hook.cc
+++ b/apt-hook/hook.cc
@@ -248,8 +248,7 @@ static void process_template_file(
          message_static_file << message_tmpl;
          message_static_file.close();
       }
-   }
-   else {
+   } else {
       remove(static_file_name.c_str());
    }
 }
@@ -260,6 +259,7 @@ static void process_all_templates(
    int esm_i_pkgs_count,
    std::string esm_i_pkgs
 ) {
+   int bytes_written;
    std::array<std::string, 4> static_file_names = {
       APT_PRE_INVOKE_APPS_PKGS_STATIC_PATH,
       MOTD_APPS_PKGS_STATIC_PATH,
@@ -313,8 +313,16 @@ static void process_all_templates(
            message_file.close();
        };
    }
-   apt_pre_invoke_msg << std::endl;
+   bytes_written = apt_pre_invoke_msg.tellp();
+   if (bytes_written > 0) {
+       // Then we wrote some content add trailing newline
+       apt_pre_invoke_msg << std::endl;
+   }
    apt_pre_invoke_msg.close();
+   if (bytes_written == 0) {
+       // We added nothing. Remove the file
+       remove(APT_PRE_INVOKE_MESSAGE_STATIC_PATH);
+   }
 
    std::ofstream motd_msg;
    motd_msg.open(MOTD_ESM_SERVICE_STATUS_MESSAGE_STATIC_PATH);
@@ -328,8 +336,16 @@ static void process_all_templates(
            message_file.close();
        };
    }
-   motd_msg << std::endl;
+   bytes_written = motd_msg.tellp();
+   if (bytes_written > 0) {
+       // Then we wrote some content add trailing newline
+       motd_msg << std::endl;
+   }
    motd_msg.close();
+   if (bytes_written == 0) {
+       // We added nothing. Remove the file
+       remove(MOTD_ESM_SERVICE_STATUS_MESSAGE_STATIC_PATH);
+   }
 }
 
 static void output_file_if_present(std::string file_name) {
```

**Pull request merged**